### ISSUE TITLE
How to customize fonts

### ISSUE THREAD

**Issue by novikoffav** [pkp/healthSciences#165]
Hi! I tried to customize fonts in typography.less but without success. I tried to combine google fonts Open Sans and Montserrat using @import url, but it does not work for me.

Would you please provide a code or some instruction how to do this?

**Comment by Vitaliy-1** [comment]
Hi @novikoffav,

In general, we are adding fonts through [PHP](https://github.com/pkp/healthSciences/blob/master/HealthSciencesThemePlugin.inc.php#L49-L53) and then assigning it to a [variable](https://github.com/pkp/healthSciences/blob/master/styles/variables.less#L27) in LESS.

I've never assigned it through import. What code you are trying to use and what errors do you see?

**Comment by novikoffav** [comment]
I tried to change fonts in PHP and in LESS as you described, but I still experience problems - it looks that browser applied some defaul fonts instead of provided by me.

In particular, I have changed a line 
'https://fonts.googleapis.com/css?family=Droid+Serif:200,200i,400,400i|Fira+Sans:300,300i,400,400i,700,700i',
onto 
'https://fonts.googleapis.com/css?family=EB+Garamond:400,400i,700,700i|Open+Sans:400,400i,700,700i&display=swap&subset=cyrillic,cyrillic-ext,latin-ext',

and I also specified fonts in Variables.less:
@font-sans:         'Open Sans', sans-serif;
@font-serif:        'EB Garamond', serif;

**Comment by Vitaliy-1** [comment]
After changing the code in PHP and in LESS, what do you see in the browser's console? Are there any errors? Does this font apply?

**Comment by novikoffav** [comment]
I do not see there any special errors. There are only two errors and three notifications in browser console, but I think they are not related to fonts (please see screenshot).
What I actually found: Before changing fonts there was an issue - when page was displayed on Cyrillic, and if the title of article was containing both cyrillic and latin words, then cyrillic words were displayed by different font which was a bit smaller than that applied to latin words. As a result in this case the whole title of article was represented by words of different sizes due to different fonts applied, as I think. After changing fonts in PHP and LESS, the whole title containing both cyrillic and latin words, now is displayed by single font and all words are of the same size **but** it is still not the font which I indicated in PHP and LESS, it is some default font like Times New Roman or something simillar.
![Screenshot_3](https://user-images.githubusercontent.com/22555461/64512148-6defeb00-d2ee-11e9-99af-6e64920c48b9.jpg)

**Comment by Vitaliy-1** [comment]
Yeah, nothing related. I've tested and except non-adjusted font sizes, new fonts are displayed correctly.
Can you confirm that you added a link to the PHP file as:
```
$this->addStyle(
				'fonts2',
				'https://fonts.googleapis.com/css?family=EB+Garamond:400,400i,700,700i%7COpen+Sans:400,400i,700,700i&display=swap&subset=cyrillic,cyrillic-ext,latin-ext',
				array('baseUrl' => '')
			);
```
and replaced variable in LESS as: 
```
@font-sans:         'Open Sans', sans-serif;
@font-serif:        'EB Garamond', serif;
```
Did you do this in a core theme files or through a child theme?
If you did it through child theme, can you show an example via GitHub? If it's modified Health Sciences theme, can you fork the repository and commit your changes there, so I can take a look at the code?

As per another question, regarding original fonts, I'm not sure why Cyrillic isn't rendered. @sssoz, I don't see `Droid Serif` fonts in the list of Google Fonts. Google refers to an external site: https://fonts.google.com/?query=droid+serif. On the other hand, the link provided in the theme is working: https://fonts.googleapis.com/css?family=Droid+Serif:200,200i,400,400i|Fira+Sans:300,300i,400,400i,700,700i. And it states, that `Droid Serif` supports Cyrillic (apparently not). Can you clarify this issue because I'm confused?

**Comment by sssoz** [comment]
@Vitaliy-1 

>  @sssoz, I don't see Droid Serif fonts in the list of Google Fonts. Google refers to an external site: https://fonts.google.com/?query=droid+serif 

This is also confusing -- it is possible that the font was once on Google Fonts (I had made sure that they were all taken from there) but the foundry (in this case, Monotype) decided to change the way it distributes it. Since it was once on there, I’m guessing they kept it on the Fonts API. 

I figured that linking to the Google Fonts API would be useful, since any updates / improvements made to the font files would be propagated on the OJS instances using that theme... but do you think it would be a better idea to load the font files as part of the theme, in the static assets directory, instead?

**Comment by sssoz** [comment]
Looks like other people have noticed: https://github.com/google/fonts/issues/1223#issuecomment-364020818

**Comment by ViRuSTriNiTy** [comment]
You could create a child theme based on health sciences, disable CDNs and add a less file with the appropriate rules to override the fonts used.

**Comment by EduardoOsquel** [comment]
How can I put a banner in the background as shown on the demo page?

https://demo.publicknowledgeproject.org/ojs3/demo/index.php/health-sciences

**Comment by webnucleo** [comment]
Not related to this theme, but I found the same problem in defining in the child-theme (Immersion Theme) the inclusion of a new google font. Didn't work. My case reported here: [https://forum.pkp.sfu.ca/t/change-fonts-child-theme/67209/2]
